Remove native_functions.yaml dependency from Sorting.cpp

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67875
* #67874
* #67721
* __->__ #66980
* #66979

Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

peterbell10 committed:
Remove native_functions.yaml dependency from Sorting.cpp

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

dagitses commented on aten/src/ATen/native/Sorting.cpp:
```diff
@@ -45,6 +45,17 @@ namespace native {
+  auto idx_dim = at::arange(0, dim_size, indices.options().dtype(at::kLong));
+  auto idx_dim_sizes = std::vector<int64_t>(indices.dim(), 1);
+  auto idx_dim_strides = std::vector<int64_t>(indices.dim(), 0);
+  idx_dim_sizes[dim] = dim_size;
```
see if this suppresses internal lint warnings:
`assert(dim < indices.dim());`

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Fix overflow for 32-bit size_t on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31856101).

peterbell10 committed:
Rebase on master on "Remove native_functions.yaml dependency from Sorting.cpp"


Differential Revision: [D31856101](https://our.internmc.facebook.com/intern/diff/D31856101)

[ghstack-poisoned]
